src/panels/config/automation/ha-automation-editor.ts

```diff
@@ -157,6 +182,9 @@ class HaAutomationEditor extends LitElement {
                 config[key] = [value];
               }
             }
+            // Drop ID from config - this is already stored in routeData and shouldn't be editable
+            // @ts-ignore
+            delete config.id;
```

Comment:
Can we just hide it from the `yaml` instead of deleting it?

Reply:
It's not possible to set it as undefined as it crashes js-yaml, but I've wrapped in a setter function that stops it being dumped